solutions/CSES Path Queries II.mdx

```diff
@@ -0,0 +1,154 @@
+---
+id: cses-2134
+source: CSES
+title: Path Queries II
+author: Dong Liu
+---
+
+# Solution
+
+This problem can solved with [Heavy Light Decomposition](/plat/hld); we can label each edge as either heavy or light. We can use a segment tree to keep track of the maximum value in each heavy chain.
+
+Now, to change the value at node $i$ to $x$, we can just update the value in the segment tree. To query the maximum value in the path from $a$ to $b$, we first find the [Lowest Common Ancestor](/gold/tree-euler#lca). Our answer would be the path from $a$ to $lca(a,b)$ and the path from $b$ to $lca(a,b)$.
```

Comment:
> Our answer would be the path from $a$ to $lca(a,b)$ and the path from $b$ to $lca(a,b)$.

It's probably better to say that we combine the two paths because the answer isn't exactly the *path*